.github/workflows/black.yml

```diff
@@ -0,0 +1,22 @@
+name: black
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  black:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+      
+      - name: Install Black
+        run: |
+          python -m pip install --upgrade pip
+          pip install black
+      
+      - name: Run Black check
```

Comment:
Why not use the action provided by `black`: https://black.readthedocs.io/en/stable/integrations/github_actions.html

Reply:
much nicer, thank you

Reply:
actually not sure if the change I just made is exactly what you had in mind, this is all relatively mysterious to me still